src/Ocelot.Provider.Consul/OcelotBuilderExtensions.cs

```diff
@@ -11,7 +12,9 @@ public static IOcelotBuilder AddConsul(this IOcelotBuilder builder)
     {
```

Comment:
Since the users should be able to implement their own `IConsulServiceBuilder`, it might be better to have `AddConsul<T>` with T, the class implementing `IConsulServiceBuilder`

Reply:
@raman-m why not `AddConsul<T>` ?

Reply:
In the proposed changes, both classes, `Consul` and `DefaultConsulServiceBuilder`, contain virtual methods. It's highly likely that developers will not override the `Consul` virtual methods. However, developers should indeed inject their own subclass of `DefaultConsulServiceBuilder`. The question at hand is the precise generic signature: should it be `AddConsul<T>` or `AddConsul<T1, T2>`? Initially, `AddConsul<T>` seems adequate. For replacing `Consul` in DI, the standard approach is to use [Custom Providers](https://ocelot.readthedocs.io/en/latest/features/servicediscovery.html#custom-providers).

🆗 Working on adding new `AddConsul<T>`...

Reply:
New method was added in commit https://github.com/ThreeMammals/Ocelot/pull/2067/commits/414d8dc3085d32028e4b9aebeee6c74845cb5d74 ✔️